crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -114,12 +114,14 @@ def versions
   def transition
     id = params[:id]       # Provisioner id
     state = params[:state] # State of node transitioning
-    name = params[:name] # Name of node transitioning
+    name = params[:name] # Name or alias of node transitioning
 
-    unless valid_transition_states.include?(state)
-      render text: "State '#{state}' is not valid.", status: 400
-    else
-      status, response = @service_object.transition(id, name, state)
+    node = Node.find_node_by_name_or_alias(name)
+
+    raise ActionController::RoutingError, "Not Found" if node.nil?
```

Comment:
I think CI doesn't pass because we actually need to not return an error if the state is `discovering`: that's how we initially create a node (so of course, `node.nil?` is true in that case). When this happens, we need to assume that `name` is indeed a name, and not an alias.

Reply:
@vuntz this case was handled previously by `valid_transition_states` ?

Reply:
Unless I misunderstand, no. Previously, we were not validating the name of the node, and therefore it was always accepted, regardless of the state we're transitioning too (which also means we could add a node directly in state ready, which is kind of wrong).